File: app/src/main/java/co/kr/sumai/GuideActivity.java
package co.kr.sumai;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;

public class GuideActivity extends AppCompatActivity {

    private static final int NUM_PAGES = 3;
    private ViewPager mPager;
    private PagerAdapter pagerAdapter;
    private LinearLayout buttonLinearLayout;
    private Button termsButton;
    private Button privacyButton;
    private Button noticesButton;

    private String[] urls = {"https://www.sumai.co.kr/terms/content", "https://www.sumai.co.kr/privacy/content", "https://www.sumai.co.kr/notices/content"};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_guide);

        buttonLinearLayout = (LinearLayout) findViewById(R.id.buttonLinearLayout);
        termsButton = (Button) findViewById(R.id.termsButton);
        privacyButton = (Button) findViewById(R.id.privacyButton);
        noticesButton = (Button) findViewById(R.id.noticesButton);
        termsButton.setTag(0);
        privacyButton.setTag(1);
        noticesButton.setTag(2);
        termsButton.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                mPager.setCurrentItem(0);
            }
        });
        privacyButton.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                mPager.setCurrentItem(1);
            }
        });
        noticesButton.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                mPager.setCurrentItem(2);
            }
        });

        mPager = (ViewPager) findViewById(R.id.guideViewPager);
        pagerAdapter = new ScreenSlidePagerAdapter(getSupportFragmentManager());
        mPager.setAdapter(pagerAdapter);
        mPager.setOffscreenPageLimit(2);

        mPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {}
            @Override
            public void onPageSelected(int position) {
                Button button;
                for(int i=0; i<3; i++) {
                    button = (Button) buttonLinearLayout.findViewWithTag(i);
                    if(position == i) {
                        buttonLinearLayout.findViewWithTag(i).setSelected(true);
                        button.setTextColor(Color.WHITE);
                    }
                    else {
                        buttonLinearLayout.findViewWithTag(i).setSelected(false);
                        button.setTextColor(Color.BLACK);
                    }
                }
            }
            @Override
            public void onPageScrollStateChanged(int state) {}
        });
    }

    @Override
    public void onStart() {
        super.onStart();

        mPager.setCurrentItem(2);
        Bundle b = getIntent().getExtras();
        int page = 0;
        if(b != null) page = b.getInt("page");
        mPager.setCurrentItem(page);
    }

    private class ScreenSlidePagerAdapter extends FragmentStatePagerAdapter {
        public ScreenSlidePagerAdapter(FragmentManager fm) {
            super(fm);
        }



        @Override
        public Fragment getItem(int position) {
            return new WebViewFragment(urls[position]);
        }

        @Override
        public int getCount() {
            return NUM_PAGES;
        }
    }
}

File: app/src/main/java/co/kr/sumai/GuideActivity.kt
package co.kr.sumai

import kotlinx.android.synthetic.main.activity_guide.*
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.OnPageChangeListener

class GuideActivity : AppCompatActivity() {
    private val urls = arrayOf("https://www.sumai.co.kr/terms/content", "https://www.sumai.co.kr/privacy/content", "https://www.sumai.co.kr/notices/content")
    override fun onCreate(savedInstanceState: Bundle?) {
        val theme = intent.getIntExtra("theme", R.style.AppVoiTheme)
        setTheme(theme)

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_guide)
        termsButton.tag = 0
        privacyButton.tag = 1
        noticesButton.tag = 2
        termsButton.setOnClickListener { guideViewPager.currentItem = 0 }
        privacyButton.setOnClickListener { guideViewPager.currentItem = 1 }
        noticesButton.setOnClickListener { guideViewPager.currentItem = 2 }
        val pagerAdapter = ScreenSlidePagerAdapter(supportFragmentManager)
        guideViewPager.adapter = pagerAdapter
        guideViewPager.offscreenPageLimit = 2
        guideViewPager.addOnPageChangeListener(object : OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
            override fun onPageSelected(position: Int) {
                var button: Button
                for (i in 0..2) {
                    button = buttonLinearLayout.findViewWithTag<View>(i) as Button
                    if (position == i) {
                        buttonLinearLayout.findViewWithTag<View>(i).isSelected = true
                        button.setTextColor(Color.WHITE)
                    } else {
                        buttonLinearLayout.findViewWithTag<View>(i).isSelected = false
                        button.setTextColor(Color.BLACK)
                    }
                }
            }

            override fun onPageScrollStateChanged(state: Int) {}
        })
    }

    public override fun onStart() {
        super.onStart()
        guideViewPager.currentItem = 2
        val b = intent.extras
        var page = 0
        if (b != null) page = b.getInt("page")
        guideViewPager.currentItem = page
    }

    private inner class ScreenSlidePagerAdapter(fm: FragmentManager?) : FragmentStatePagerAdapter(fm!!) {
        override fun getItem(position: Int): Fragment {
            return WebViewFragment(urls[position])
        }

        override fun getCount(): Int {
            return NUM_PAGES
        }
    }

    companion object {
        private const val NUM_PAGES = 3
    }
}